tests/integration/test_reqs.py

```diff
@@ -0,0 +1,56 @@
+import pytest
+
+
+@pytest.mark.requirements
+def test_reqs_generates_requirements_from_lockfile(PipenvInstance):
+    with PipenvInstance(chdir=True) as p:
+        packages = ('requests', '2.14.0')
+        dev_packages = ('flask', '0.12.2')
+        with open(p.pipfile_path, 'w') as f:
+            contents = f"""
+            [packages]
+            {packages[0]}= "=={packages[1]}"
+            [dev-packages]
+            {dev_packages[0]}= "=={dev_packages[1]}"
+            """.strip()
+            f.write(contents)
+        p.pipenv('lock')
+        c = p.pipenv('reqs')
+        d = p.pipenv('reqs --dev')
+        assert c.returncode == 0
+        assert d.returncode == 0
+
+        assert f'{packages[0]}=={packages[1]}' in c.stdout
+        assert f'{dev_packages[0]}=={dev_packages[1]}' not in c.stdout
+
+        assert f'{packages[0]}=={packages[1]}' in d.stdout
+        assert f'{dev_packages[0]}=={dev_packages[1]}' in d.stdout
+
+
+@pytest.mark.requirements
+def test_reqs_generates_requirements_from_lockfile_multiple_sources(PipenvInstance):
+    with PipenvInstance(chdir=True) as p:
+        packages = ('requests', '2.14.0')
+        dev_packages = ('flask', '0.12.2')
+        with open(p.pipfile_path, 'w') as f:
+            contents = f"""
+            [[source]]
+            name = "pypi"
+            url = "https://pypi.org/simple"
+            verify_ssl = true
+            [[source]]
+            name = "other_source"
+            url = "https://some_other_source.org"
+            verify_ssl = true
+            [packages]
+            {packages[0]}= "=={packages[1]}"
+            [dev-packages]
+            {dev_packages[0]}= "=={dev_packages[1]}"
+            """.strip()
+            f.write(contents)
+        p.pipenv('lock')
+        c = p.pipenv('reqs')
+        assert c.returncode == 0
+
+        assert '-i https://pypi.org/simple' in c.stdout
```

Comment:
Users should be aware that generating a requirements.txt with pinned indexes will result in a pip command that will not fully respect the restricted indexes.   This was something that came up recently where we now have patched the pip internals within pipenv to ensure restricted indexes are obeyed, but this is another case where the end user only gets that benefit if they use pipenv to sync or install --deploy their dependencies, because pip itself will prefer a package in pypi and just use extra-indexes to search for things not in pypi, basically (it is probably a little more complicated than that but in a nutshell).

Reply:
I also think that this should not be the main route, but if I don't add the indexes and you use multiple, the requirements file will fail completely. Anything required on my part currently?

Reply:
I think you're good, I just was calling that out because the test reminded me of that fact.  I kicked off the test CI.